### PR TITLE
docs: fix events in organisms and templates docs 

### LINKED ISSUE #2115: [BUG] Docs: check and fix events
**Describe the bug**

- Check events, if they show properly  in Storybook docs.
- Fix bugs.
- Add descriptions when they are needed.

**How to reproduce**

**Expected behavior**

**Actual behavior**
What actual happened.

**Code examples**
Sample code (or your configuration) to help us reproduce the bug faster or to understand your use case.
```
//Write code here
```
Or add code snippets (Code sandbox)

**Screenshots**

**Devices (please complete the following information):**
 - Device: [e.g. iPhone6, Desktop]
 - Browser: [e.g chrome, safari][version]

**Additional context**
Anything you would like to add but was not sure which section it fits. For example: tech stack used (Nuxt, Vue, Gridsome, React, etc)

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue

Closes #2115 

# Scope of work
Add descriptions for events and fix events in storybook.
Also fix for events in:
-  MegaMenu  - click:back event 
-  Header - keyup.enter event

# Screenshots of visual changes

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/organisms/SfGroupedProduct/SfGroupedProduct.stories.js [hunk 1]
```diff
@@ -432,7 +443,7 @@ export const UseTitleSlot = (args, { argTypes }) => ({
   >
     <SfGroupedProductItem
       :qty="productQty"
-      @input="productQty = $event"
+      @input="inputHandler"
```
Comment (aniamusial) [inline packages/vue/src/components/organisms/SfGroupedProduct/SfGroupedProduct.stories.js:446]: question: in file 56 f.e you are also passing the $event to inputHandler - both are ok, but maybe we should keep consistency?

File: packages/vue/src/components/organisms/SfHeader/SfHeader.vue [hunk 2]
```diff
@@ -35,7 +35,7 @@
               aria-label="Search"
               class="sf-header__search"
               @input="$emit('change:search', $event)"
-              @enter="$emit('enter:search', $event)"
+              @keyup.enter="$emit('enter:search', $event)"
```
Comment (aniamusial) [inline packages/vue/src/components/organisms/SfHeader/SfHeader.vue:38]: praise: nice catch!

**Review by justyna-13 — COMMENTED** [review]
Story of SfArrcorion with buttons doesn't work

**Review by aniamusial — APPROVED** [review]

**Review by justyna-13 — COMMENTED** [review]
Can you check also this order?

Screencast-from-12-03-2021-01_50_47-PM

**Comment by AdamPawlinski** [comment]
@justyna-13 I hope that it's working now :)

**Review by justyna-13 — APPROVED** [review]

**Pull request merged**